cmd/tsbs_generate_queries/databases/timescaledb/devops_test.go

```diff
@@ -152,3 +155,324 @@ func TestDevopsGetSelectClausesAggMetrics(t *testing.T) {
+		"max(usage_nice) as max_usage_nice, max(usage_iowait) as max_usage_iowait, max(usage_irq) as max_usage_irq, " +
+		"max(usage_softirq) as max_usage_softirq, max(usage_steal) as max_usage_steal, max(usage_guest) as max_usage_guest, " +
+		`max(usage_guest_nice) as max_usage_guest_nice
+        FROM cpu
+        WHERE (hostname = 'host_9') AND time >= '1970-01-01 02:16:22.646325 +0000' AND time < '1970-01-01 10:16:22.646325 +0000'
+        GROUP BY hour ORDER BY hour`
+	rand.Seed(123) // Setting seed for testing purposes.
+	s := time.Unix(0, 0)
+	e := s.Add(devops.DoubleGroupByDuration).Add(time.Hour)
+
+	d := NewDevops(s, e, 10)
+
+	q := d.GenerateEmptyQuery()
+	d.MaxAllCPU(q, 1)
+	verifyQuery(t, q, expectedHumanLabel, expectedHumanDesc, expectedHypertable, expectedSQLQuery)
+}
+
+func TestLastPointPerHost(t *testing.T) {
+	cases := []struct {
+		desc               string
+		useJSON            bool
+		useTags            bool
+		expectedHumanLabel string
+		expectedHumanDesc  string
+		expectedHypertable string
+		expectedSQLQuery   string
+	}{
+		{
+			desc:               "no JSON or tags",
+			expectedHumanLabel: "TimescaleDB last row per host",
+			expectedHumanDesc:  "TimescaleDB last row per host",
+			expectedHypertable: "cpu",
+			expectedSQLQuery:   "SELECT DISTINCT ON (hostname) * FROM cpu ORDER BY hostname, time DESC",
+		},
+		{
+			desc:               "use JSON",
+			useJSON:            true,
+			expectedHumanLabel: "TimescaleDB last row per host",
+			expectedHumanDesc:  "TimescaleDB last row per host",
+			expectedHypertable: "cpu",
+			expectedSQLQuery: "SELECT DISTINCT ON (t.tagset->>'hostname') * FROM tags t INNER JOIN LATERAL(SELECT * " +
+				"FROM cpu c WHERE c.tags_id = t.id ORDER BY time DESC LIMIT 1) AS b ON true ORDER BY t.tagset->>'hostname', b.time DESC",
+		},
+		{
+			desc:               "use tags",
+			useTags:            true,
+			expectedHumanLabel: "TimescaleDB last row per host",
+			expectedHumanDesc:  "TimescaleDB last row per host",
+			expectedHypertable: "cpu",
+			expectedSQLQuery: "SELECT DISTINCT ON (t.hostname) * FROM tags t INNER JOIN LATERAL(SELECT * FROM cpu c " +
+				"WHERE c.tags_id = t.id ORDER BY time DESC LIMIT 1) AS b ON true ORDER BY t.hostname, b.time DESC",
+		},
+		{
+			desc:               "enable JSON and tags but use tags",
+			useJSON:            true,
+			useTags:            true,
+			expectedHumanLabel: "TimescaleDB last row per host",
+			expectedHumanDesc:  "TimescaleDB last row per host",
+			expectedHypertable: "cpu",
+			expectedSQLQuery: "SELECT DISTINCT ON (t.hostname) * FROM tags t INNER JOIN LATERAL(SELECT * FROM cpu c " +
+				"WHERE c.tags_id = t.id ORDER BY time DESC LIMIT 1) AS b ON true ORDER BY t.hostname, b.time DESC",
+		},
+	}
+
+	rand.Seed(123) // Setting seed for testing purposes.
+	s := time.Unix(0, 0)
+	e := s.Add(devops.DoubleGroupByDuration).Add(time.Hour)
```

Comment:
this doesn't need to be `Add(devops.DoubleGroupByDuration)`, I actually just meant that for the double groupby queries :)